Modules/_pickle.c

```diff
@@ -821,22 +821,18 @@ _PyMemoTable_Lookup(PyMemoTable *self, PyObject *key)
 
 /* Returns -1 on failure, 0 on success. */
 static int
-_PyMemoTable_ResizeTable(PyMemoTable *self, Py_ssize_t min_size)
+_PyMemoTable_ResizeTable(PyMemoTable *self, size_t min_size)
 {
     PyMemoEntry *oldtable = NULL;
     PyMemoEntry *oldentry, *newentry;
-    Py_ssize_t new_size = MT_MINSIZE;
+    size_t new_size = MT_MINSIZE;
     Py_ssize_t to_process;
 
     assert(min_size > 0);
 
     /* Find the smallest valid table size >= min_size. */
-    while (new_size < min_size && new_size > 0)
+    while (new_size < min_size)
```

Comment:
Wouldn't this cause an infinite loop if `min_size` > `PY_SSIZE_T_MAX`?

Reply:
Yes, I will fix.